shell/platform/embedder/embedder.cc

```diff
@@ -565,6 +565,43 @@ static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
   return surface;
 }
 
+static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
+    GrDirectContext* context,
+    const FlutterBackingStoreConfig& config,
+    const FlutterMetalBackingStore* metal) {
+#ifdef SHELL_ENABLE_METAL
+  GrMtlTextureInfo texture_info;
+  texture_info.fTexture.reset(metal->texture.texture);
```

Comment:
This assumes transfer-in ownership. Let's make it (and document) transfer-none so its easier to reason about in case of error.